### ISSUE TITLE
Number of initially collapsed/unfolded comments should be made configurable

### ISSUE THREAD

**Issue by wiwie** [friendica/friendica#8421]
Currently it seems that under posts always two comments are unfolded and there is no way to collapse them.

If comments are long, this can take up a lot of space on mobile.

Also, with some posts I'm not interested in the comments, so I'd like if all the comments were collapsed by default.

And this could be made a setting for the user how comments should initially be shown: all, none, some (max x).

**Comment by wiwie** [comment]
Any thoughts on this? Would you be willing to merge a PR with this feature in?

**Comment by MrPetovan** [comment]
I believe it's a good idea. However the current system isn't ideal because it cuts off at 3 **first-level comments** which can actually turn out to be many more if these comments got any sub-comments. So replacing it by 2, 1 or 5 by the user wouldn't change this weakness.

**Comment by wiwie** [comment]
Yes I saw the on my wall as well. For now as a temporary solution I'll change line
```
$result['children'][$nb_children - 3]['comment_lastcollapsed'] = true;
```
In Post.php to

```
$result['children'][$nb_children]['comment_lastcollapsed'] = true;
```
I.e. not showing any comments by default.
In my opinion this is already way easier on the eyes.

One "hacky" way could be to apply the same kind of limit height functionality that was recently merged with the showmore_dyn add-on. Meaning, actually expanding the whole comment tree but visually cutting it off.

Second approach I could think of is to have collapse/expand buttons at each level of the hierarchy after each comment that has sub comments. Disadvantage of this could be that it quickly grows tiresome that one needs to expand too often. Probably not they best solution.

I'll need some more time to think about the options.

**Comment by MrPetovan** [comment]
Please let me know what you come up with.

**Comment by wiwie** [comment]
I didn't forget about this issue and keep thinking about it once in a while, but haven't really come up with anything smart.

I do think that it's ok to have comments shown by default. But what annoys me about the current solution most is that it is not possible to hide all comments, not even by clicking a button. Toggling the existing button shows more than X or shows X comments. But X>0. If X=0 or if it was configurable in the user/theme settings, the current logic would actually be fine by me.

Reddit has btw implemented my first suggested solution from my last post here for comments to a post. It seems they cutoff the div with the comments at some prespecified height and append a toggle button to it "show all XYZ comments".

**Comment by SpcCw** [comment]
I tried hiding comments by default [as suggested above](https://github.com/friendica/friendica/issues/8421#issuecomment-601937637) and I love it!

If there are no better solutions maybe we can have this as user option? 

So far it works great ("show more" link should be reworded of course). It is such a relief when you have some noisy contacts leaving l-o-o-o-n-g comment threads...

**Comment by vinzv** [comment]
Digging this one up as I came across the uncommon collapsing behavior myself. Thanks for the hacky solution, btw.

Any chances we get a toggle or config setting for this? I would be pretty happy with the "collapse all" approach.

**Comment by annando** [comment]
I think that we should rework this part completely. We have to rework this whole part to load the comments dynamically.
